week3/homework/src/actions.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const { readFile, writeFile } = require('fs');
+const { promisify } = require('util');
+const uuid = require('uuid/v4');
+const read = promisify(readFile);
+const write = promisify(writeFile);
+const FILE_PATH = 'todo.json';
+const list = {};
+
+
+const fileContent = (todo) => {
+    return write(FILE_PATH, JSON.stringify(todo, null, 2));
+};
+
+const createTodo = async (todo) => {
+    if (!todo.description) throw new Error('Description is required');
+
+    const todoList = await readTheList();
+    const id = uuid();
+    list[id] = todo;
+    list[id].done = false;
+    todoList.push(list);
+    await fileContent(todoList);
+    return todoList;
+};
+
+const readTheList = () => {
+    return read(FILE_PATH, 'utf-8')
+        .then(JSON.parse)
+        .catch(() => ([]));
+};
+
+const updateTodo = async (id, newTodo) => {
+    const todoList = await readTheList();
+    try {
+        const foundItem = todoList.find(todo => todo[id]);
+        foundItem[id].description = newTodo.description;
+        await fileContent(todoList);
+        return todoList;
+    } catch (error) {
+        throw new Error('Item cannot be found.');
+    }
+};
+
+const markAsDoneOrNotDone = async (id, boolean) => {
+    const todoList = await readTheList();
+
+    try {
+        const foundItem = todoList.find(todo => todo[id]);
+        foundItem[id].done = boolean;
+        await fileContent(todoList);
+        return todoList;
+    } catch (error) {
```

Comment:
This error handling doesn't really distinguish between item not being found and file failing to be written.

Reply:
That's right, the thing is at first I used the catch() method in fileContent () function, to display a message if something goes wrong. But the console would not show that error message and would show specific error messages based on the problem. So I thought maybe catch is not necessary. Can you maybe explain that?